### PR TITLE
Document the BonnyCI Workflow

### PR DESCRIPTION
**Pull request by eggshell**
* Adds documentation and a diagram for how automated testing and the
   BonnyCI process apply to a consumer's repository once added.

* Gives a step-by-step explanation of how pull requests are tested
   and merged with BonnyCI.

Resolves: BonnyCI/projman#84

Signed-off-by: Cullen Taylor <CullenTaylor@outlook.com>

### CODE REVIEW

**Comment by eggshell** [comment]
Closing in favor of #27

**Review by ConsoleCatzirl — APPROVED** [review]
Looks great

File: end_users/use/README.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+If a pull request fails the check queue, then it will not be able to move on to further testing. The developer will make changes to the pull request until it passes the check queue. Concurrently, core contributors of the project will be reviewing pull requests. Core contributors confirm pull requests' technical value and guideline conformity.
+
+Now we have a pull request that has core contributor approval and successful check queue results. That pull request will automatically move on to the "gate" queue. The gate queue collects all pull requests in this state and tests them together in their theoretical merge order. If any pull request in the gate queue fails, BonnyCI reports it on the pull request's conversation page. The pull requests that remain in the gate queue are re-tested together in their new theoretical merge order. This process repeats until no failures are present. Any pull requests which pass the gate queue are automatically merged. This ensures that what gets tested is what gets merged, and is new functionality to GitHub users.
```
Comment (omgjlk) [inline end_users/use/README.md:15]: This "Now we have" seems awkward, and could use rephrasing, but that can be a later update.

File: end_users/use/README.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+## Pull Request Lifecycle
+
+1. New pull request is opened against your repository.
+2. BonnyCI places pull request on top of existing repository codebase and tested independent of any other open pull requests.
+3. Core contributors to the project review the pull request. If changes are requested, the pull request author will make them and amend the commit.
+4. Core contriubtors approve the pull request and add the `approved` label to the pull request.
+5. BonnyCI places all pull requests with the `approved` label on top of the existing repository codebase in the order they were submitted. If any pull request fails the testing, BonnyCI will kick that pull request out and test the remaining changes together. This testing process will recurse until the build passes.
```
Comment (omgjlk) [inline end_users/use/README.md:23]: The use of the `approved` label will change when we implement triggering/gating on review state instead. Just commenting to keep us aware of that.

**Review by omgjlk — APPROVED** [review]
Good stuff, merging!

File: end_users/use/README.md [hunk 3]
```diff
@@ -0,0 +1,30 @@
+# Use
```
Comment (mlangbehn) [inline end_users/use/README.md:1]: Calling this "use" is a bit confusing. Can we title something different? How about "workflow"?

File: end_users/README.md [hunk 4]
```diff
@@ -3,13 +3,18 @@
...
 * [Bug Reports](#bug-reports)
 * [Feature Requests](#feature-requests)
 
 ## Setup
 
 Check out the [setup documentation](setup/README.md) for instructions on how to get started with BonnyCI!
 
+## Using BonnyCI
+
+Once you have your repo set up to use BonnyCI, check out the [use documentation](use/README.md) for information on the BonnyCI process.
```
Comment (mlangbehn) [inline end_users/README.md:16]: Same comment as in the other file about possibly changing "use" to "workflow" (or something other than "use").

**Pull request merged**